src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -465,11 +465,25 @@ private Device GetDeviceFromRequest(ValidatedRequest request)
                     CoreHelpers.CustomProviderName(type));
                 if (type == TwoFactorProviderType.Duo)
                 {
-                    return new Dictionary<string, object>
+                    // DUO SDK v4 Update: Duo-Redirect
+                    if (FeatureService.IsEnabled(FeatureFlagKeys.DuoRedirect, CurrentContext))
                     {
-                        ["Host"] = provider.MetaData["Host"],
-                        ["Signature"] = token
-                    };
+                        CurrentContext.HttpContext.Request.Headers.TryGetValue("X-Bitwarden-Client-Name", out var bitwardenClientName);
+                        return new Dictionary<string, object>
+                        {
+                            ["Host"] = provider.MetaData["Host"],
+                            ["Duo2faInitiatingClient"] = bitwardenClientName,
+                            ["AuthUrl"] = token
```

Comment:
Can we append the client name to the auth url before sending down to client?

Reply:
I assumed we could use the `Dui2faInitiatingClient` to more directly track the client name instead of having to parse the AuthUrl.

Reply:
appended to the redirect.